prow/resultstore/uploader.go

```diff
@@ -38,6 +38,10 @@ func NewUploader(client *Client) *Uploader {
 func (u *Uploader) Upload(ctx context.Context, log *logrus.Entry, p *Payload) error {
 	inv, err := p.Invocation()
 	if err != nil {
+		if writer.PermanentError(err) {
+			// Return nil error to avoid outer retries.
```

Comment:
Nit: perhaps "NOP if permanent errors are encountered, because there's no point in retrying" is a better phrasing? The wording "Return nil error" doesn't add much value because it directly reflects what's happening in the code.

Reply:
I updated the function documentation to clarify why some errors aren't returned.

Also note I had the error check on the wrong case. But that one also should not be retried.